spinalcordtoolbox/scripts/sct_analyze_lesion.py

```diff
@@ -311,6 +359,354 @@ def _measure_axial_damage_ratio(self, im_data, p_lst, idx):
+
+            # Save info for plotting
+            self.tissue_bridges_plotting_data[lesion_idx][sagittal_slice, 'dorsal'] = (
+                df_temp)[df_temp['axial_slice'] == min_dorsal_bridge_width_slice]
+            self.tissue_bridges_plotting_data[lesion_idx][sagittal_slice, 'ventral'] = (
+                df_temp)[df_temp['axial_slice'] == min_ventral_bridge_width_slice]
+
+            # Get the width of the tissue bridges in mm
+            # NOTE: the orientation is RPI (because we reoriented the image to RPI using orient2rpi()); therefore
+            # p_lst[0] is the pixel size in the R-L direction, p_lst[1] is the pixel size in the A-P direction, and
+            # p_lst[2] is the pixel size in the S-I direction.
+            # Since we are computing dorsal and ventral tissue bridges, we use p_lst[1] (A-P direction)
+            # NOTE: we use np.cos(self.angles[SLICE]) to correct for the angle of the spinal cord with respect to the
+            # slice
+            dorsal_bridge_width_mm = (float(df_temp[df_temp['axial_slice'] == min_dorsal_bridge_width_slice]
+                                            ['dorsal_bridge_width'] * p_lst[1]) *
+                                      np.cos(self.angles[min_dorsal_bridge_width_slice]))
+            ventral_bridge_width_mm = (float(df_temp[df_temp['axial_slice'] == min_ventral_bridge_width_slice]
+                                             ['ventral_bridge_width'] * p_lst[1]) *
+                                       np.cos(self.angles[min_ventral_bridge_width_slice]))
+            total_bridge_width_mm = dorsal_bridge_width_mm + ventral_bridge_width_mm
+
+            # Save the minimum tissue bridges
+            self.measure_pd.loc[idx, f'slice_{str(sagittal_slice)}_dorsal_bridge_width [mm]'] = dorsal_bridge_width_mm
+            self.measure_pd.loc[idx, f'slice_{str(sagittal_slice)}_ventral_bridge_width [mm]'] = ventral_bridge_width_mm
+            self.measure_pd.loc[idx, f'slice_{str(sagittal_slice)}_total_bridge_width [mm]'] = total_bridge_width_mm
+            printv(f'  Sagittal slice {sagittal_slice}, Minimum dorsal tissue bridge width: '
+                   f'{np.round(dorsal_bridge_width_mm, 2)} mm (axial slice {min_dorsal_bridge_width_slice})',
+                   self.verbose, type='info')
+            printv(f'  Sagittal slice {sagittal_slice}, Minimum ventral tissue bridge width: '
+                   f'{np.round(ventral_bridge_width_mm, 2)} mm (axial slice {min_ventral_bridge_width_slice})',
+                   self.verbose, type='info')
+            printv(f'  Sagittal slice {sagittal_slice}, Total tissue bridge width: '
+                   f'{np.round(total_bridge_width_mm, 2)} mm', self.verbose, type='info')
+
+    def _plot_tissue_bridges(self, im_lesion_data, p_lst, label_lst):
+        """
+        Plot the tissue bridges.
+        The figure has one row per lesion and one column per sagittal slice containing the lesion.
```

Comment:
> would it be possible to create the QC report using only the values found in lesion_analysis.xls

This is an excellent point, actually! After removing the plotting of the line segments (https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/4489/commits/7a363b1b95030d1b6e84951d4d7a186f04ca9b0e), we indeed need just sagittal slice number and tissue bridges widths. All of this is already outputted into `lesion_analysis.xls`.

(For context: `axial_slice` and `lesion_indices` were used mainly to draw the line segments. After removing the line segments, these two variables are now used only to [place the text](https://github.com/spinalcordtoolbox/spinalcordtoolbox/blob/83f2990e10fd7bd7d7a66efb3efa9a956ca7295b/spinalcordtoolbox/scripts/sct_analyze_lesion.py#L537-L554). Which can be done easily in another way, though. For example, based on the lesion center of mass.)

Reply:
Before we go ahead and make the changes, I'm just curious: Why does the figure use the rounded values (3.0mm, 2.0mm) but the xls prints the angle-corrected values (2.997796, 1.998531).

We should probably display the angle-corrected values, right?

Reply:
Good point! 
The figure uses `np.round`: https://github.com/spinalcordtoolbox/spinalcordtoolbox/blob/42c77503619dced05be5af770276b560740fe5f7/spinalcordtoolbox/scripts/sct_analyze_lesion.py#L554

while `self.measure_pd` stores (and saves to XLS) floats:
https://github.com/spinalcordtoolbox/spinalcordtoolbox/blob/42c77503619dced05be5af770276b560740fe5f7/spinalcordtoolbox/scripts/sct_analyze_lesion.py#L230 

<img width="1052" alt="image" src="https://github.com/user-attachments/assets/05f3a27a-ba2e-450d-8b84-1ab31a5d9535">

I checked publications (for example Table 1 in [this paper](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6815206/)) that use tissue bridges, and I believe we can stick with two decimals for both the figure and XLS file

Reply:
Perfect! I'll go ahead and implement the QC report, then. :)

Reply:
The QC report has been added! It now accepts `measure_pd` as the only input (besides the images themselves), which removes the need to access `tissue_bridges_plot_dict`, `p_lst`, `angles`, etc.

Reply:
Awesome! Thank you, @joshuacwnewton! 💯 I also love your commit messages, making it easy to understand the changes.

Reply:
I tested the updated PR, and I could successfully generate the QC for the single lesion. When testing on an image with multiple lesions, I get the following error: 

```python
Traceback (most recent call last):
  File "/Users/valosek/code/sct_nnunet/spinalcordtoolbox/scripts/sct_analyze_lesion.py", line 873, in <module>
    main(sys.argv[1:])
  File "/Users/valosek/code/sct_nnunet/spinalcordtoolbox/scripts/sct_analyze_lesion.py", line 840, in main
    sct_analyze_lesion(
  File "/Users/valosek/code/sct_nnunet/spinalcordtoolbox/reports/qc2.py", line 468, in sct_analyze_lesion
    axes[idx_row, idx_col].text(min(np.where(slice_lesion)[0]) - 3,
ValueError: min() arg is an empty sequence
```

Reply:
Okay, this was an easy fix :-) Fixed in https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/4489/commits/2df1fa019282fe12064dd820022533c963d7bb33; see the commit message for details.

Reply:
Two last things:
1. we should probably add a test for multiple lesions
2. do we want to generate the QC in default (as done now)? Or maybe it would be better to generate the QC only when `sct_analyze_lesion.py -qc` is used.

Reply:
Good idea! Let's go with 2., since that seems to be the default behavior of most scripts. (I'm not quite sure why I did it this way, aha.)

Reply:
Fixes added!

Reply:
Thank you! Tested, and the code works fine with the `-qc` flag.